src/amo/components/CollectionList/index.js

```diff
@@ -0,0 +1,173 @@
+  currentUsername: string | null,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  isLoggedIn: boolean,
+  loadingUserCollections: boolean,
+  location: ReactRouterLocation,
+|};
+
+export class CollectionListBase extends React.Component<Props> {
+  componentDidMount() {
+    const {
+      collections,
+      currentUsername,
+      dispatch,
+      errorHandler,
+      loadingUserCollections,
+    } = this.props;
+
+    if (currentUsername && !loadingUserCollections && !collections) {
+      dispatch(
+        fetchUserCollections({
+          errorHandlerId: errorHandler.id,
+          username: currentUsername,
+        }),
+      );
+    }
+  }
+
+  renderCollections(header: string) {
+    const { i18n, collections } = this.props;
+    const noCollectionsText = i18n.gettext('You do not have any collections.');
+
+    const collectionElements = [];
+
+    if (collections) {
+      collections.forEach((collection) => {
+        const { authorUsername, id, name, numberOfAddons, slug } = collection;
+        collectionElements.push(
+          <UserCollection
+            authorUsername={authorUsername}
+            id={id}
+            name={name}
+            numberOfAddons={numberOfAddons}
+            slug={slug}
+          />,
+        );
+      });
+    } else {
+      // Create 4 "loading" components.
+      for (let count = 0; count < 4; count++) {
+        collectionElements.push(<UserCollection id={count} loading />);
+      }
+    }
+
+    const footer = collectionElements.length ? null : noCollectionsText;
+
+    return (
+      <Card
+        className={makeClassName('CardList', 'CollectionList-list')}
+        footer={footer}
+        header={header}
+      >
+        {collectionElements.length && (
+          <ul className="CollectionList-listing">{collectionElements}</ul>
+        )}
+      </Card>
+    );
+  }
+
+  render() {
+    const { i18n, isLoggedIn, location } = this.props;
+    const myCollectionsHeader = i18n.gettext('My Collections');
+
+    return (
+      <div className="CollectionList">
+        <div className="CollectionList-wrapper">
+          <Card className="CollectionList-info" header={myCollectionsHeader}>
+            {!isLoggedIn ? (
+              <AuthenticateButton
+                noIcon
+                location={location}
+                logInText={i18n.gettext('Log in to view your collections')}
+              />
+            ) : (
+              <React.Fragment>
+                <p className="CollectionList-info-text">
+                  {i18n.gettext(`Collections make it easy to keep track of favorite
+                    add-ons and share your perfectly customized browser with others.`)}
+                </p>
+                <Button
+                  buttonType="action"
+                  className="CollectionList-create"
+                  puffy
+                  to="/collections/add/"
+                  type="button"
```

Comment:
I don't think we should be adding this `type` prop, in part because the style is broken in Chrome:

![screen shot 2018-06-25 at 14 22 58](https://user-images.githubusercontent.com/217628/41850031-77f31cf6-7883-11e8-9c8f-3eccd2277d84.png)